### PR TITLE
mutate_exists for StorageValue with ValueQuery

### LINKED ISSUE #13133: mutate_exists for StorageValue with ValueQuery
I use ValueQuery so it creates a default value for empty key, which is handy. But I still want to delete it with `mutate_exists`.

### PR DESCRIPTION
**Pull request by muraca**
closes #13133 

polkadot address: 12poSUQPtcF1HUPQGY3zZu2P8emuW9YnsPduA4XG3oCEfJVp

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2322786

**Review by kianenigma — APPROVED** [review]

**Comment by bkchr** [comment]
bot merge

**Comment by shawntabrizi** [comment]
@muraca can you please include your polkadot address in your PR description.

See https://github.com/paritytech/substrate-tip-bot

**Review by shawntabrizi — APPROVED** [review]

File: frame/support/src/lib.rs [hunk 1]
```diff
@@ -874,6 +874,7 @@ pub mod tests {
 
 	decl_storage! {
 		trait Store for Module<T: Config> as Test {
+			pub Value get(fn value): u64;
```
Comment (ggwpez) [inline frame/support/src/lib.rs:877]: One tests fails now so you have to fix the hard-coded metadata https://github.com/paritytech/substrate/blob/c19838924ed941de28ea9d6ec25a74ccf207bc65/frame/support/src/lib.rs#L1306

Comment (ggwpez) [reply frame/support/src/lib.rs:877]: Or you define it in your test like:  
```rust
#[frame_support::storage_alias]
pub type Value = StorageValue<Prefix, u32, ValueQuery>;
```

**Review by ggwpez — APPROVED** [review]

File: frame/support/src/lib.rs (hunk 1 above)
Comment (muraca) [reply frame/support/src/lib.rs:877]: Done, I hope it's fine now

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2328124

**Comment by bkchr** [comment]
Still not compiling.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2328169

**Comment by bkchr** [comment]
bot merge

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2328222

File: frame/support/src/lib.rs [hunk 2]
```diff
@@ -1257,6 +1313,13 @@ pub mod tests {
 		PalletStorageMetadata {
 			prefix: "Test",
 			entries: vec![
+				StorageEntryMetadata {
+					name: "Value",
+					modifier: StorageEntryModifier::Default,
+					ty: StorageEntryType::Plain(scale_info::meta_type::<u32>()),
```
Comment (ggwpez) [inline frame/support/src/lib.rs:1319]:
```suggestion
					ty: StorageEntryType::Plain(scale_info::meta_type::<u64>()),
```

**Comment by ggwpez** [comment]
bot merge

**Pull request merged**